dangerfile.ts

```diff
@@ -1,10 +1,52 @@
-import { warn, danger } from 'danger'
-
-// Warn about creating new JS files
-const jsFiles = danger.git.created_files.filter(f => f.endsWith('.js'))
-if (jsFiles.length) {
-  const files = danger.github.utils.fileLinks(jsFiles)
-  warn(
-    `Please don't include .js files, we want to be using TypeScript found: ${files}.`
+import { warn, danger } from "danger"
+import { getBreakingChanges } from "./scripts/validate_schemas"
+import { BreakingChange } from "graphql"
+
+// tslint:disable-next-line:no-default-export
+export default async () => {
+  // Warn about creating new JS files
+  const jsFiles = danger.git.created_files.filter(
+    f => f.includes("src") && f.endsWith(".js")
   )
+  if (jsFiles.length) {
+    const files = danger.github.utils.fileLinks(jsFiles)
+    warn(
+      `Please don't include .js files, we want to be using TypeScript found: ${files}.`
+    )
+  }
+
+  // Breaking change check for Metaphysics production when deploying
+  if (danger.github.pr.base.ref === "release") {
```

Comment:
I know a conversation topic during our recent platform practice meeting was checking for "dependencies in prod" before code hits staging as well, as such, ensuring that we never have staging code that _shouldn't_ be deployed production.

Thoughts on updating this to run on PRs to `master` -- providing the staging check?

Reply:
I'm wary that this will "break" builds that have nothing to do with metaphysics, so happy to do this as-is now and can look at making a tighter loop for when people either update  reaction or other builds